lib/publiccloud/instance.pm

```diff
@@ -203,10 +203,57 @@ sub wait_for_guestregister
         }
         sleep 1;
     }
+
     $self->upload_log('/var/log/cloudregister', log_name => $autotest::current_test->{name} . '-cloudregister.log');
     die('guestregister didn\'t end in expected timeout=' . $args{timeout});
 }
 
+=head2 check_guestregister
+
+    check_guestregister();
+
+Check that the registration state of newly created guest match our requirements.
+https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt
+=cut
+sub check_guestregister
+{
+    my ($self) = @_;
+
+    if (is_byos) {
+        $self->run_ssh_command(cmd => '! sudo zypper lr', timeout => 6000);
```

Comment:
why is this needed? If we want to make sure repos are not defined, why don't we use the same approach as below?
```
if ($self->run_ssh_command(cmd => '! sudo zypper lr', timeout => 6000, proceed_on_failure => 1, quiet => 1) !~ /no repos defined/) {
```
ofc, not that output message :)

Reply:
In [SUSE-Enceladus/cloud-regionsrv-client/integration_test-process.txt](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt) is required both.

Reply:
I used `rc_only` but it did not really work for me. I'll try once more.

Reply:
@grisu48 @asmorodskyi this is not true. See `testapi::script_output` [documentation](http://open.qa/api/testapi/#_script_output):
> Executing script inside SUT with bash -eox (in case of serial console with bash -eo) and directs stdout (not' stderr!) > to the serial console and returns the output if' the script exits with 0. Otherwise the test is set to failed.

Reply:
Thanks for the clarification, then your code is correct :+1: